apiserver/facade/interface.go

```diff
@@ -160,6 +160,10 @@ type Context interface {
 
 	// Raft returns a lease context for managing raft.
 	Raft() RaftContext
+
+	// CharmhubHTTPClient returns an HTTP client to use for sending Charmhub
+	// API requests.
+	CharmhubHTTPClient() HTTPClient
```

Comment:
As we add more different clients, this will bloat.
Instead can we have a generic `HTTPClientPoolpurpose string)` or something. ie a single method where you ask for a http client for a given purpose. The purpose should be typed and well known values done as consts etc.